src/aiida/orm/nodes/data/remote/base.py

```diff
@@ -185,3 +194,159 @@ def _validate(self):
+
+    def get_size_on_disk(
+        self,
+        relpath: Path | None = None,
+        method: str = 'du',
+        return_bytes: bool = False,
+    ) -> int | str:
+        """Connects to the remote Computer of the `RemoteData` object and returns the total size of a file or a
+        directory at the given `relpath` in a human-readable format.
+
+        :param relpath: File or directory path for which the total size should be returned, relative to
+            ``self.get_remote_path()``.
+        :param method: Method to be used to evaluate the directory/file size (either ``du`` or ``lstat``).
+        :param return_bytes: Return the total byte size is int, or in human-readable format.
+
+        :raises FileNotFoundError: If file or directory does not exist anymore on the remote ``Computer``.
+        :raises NotImplementedError: If a method other than ``du`` or ``lstat`` is selected.
+
+        :return: Total size of given file or directory.
+        """
+
+        from aiida.common.utils import format_directory_size
+
+        # Initialize it here and only return if set. Avoid repeating the code in the `else` of the try-except-else
+        # blocks for each method
+        total_size: int | None = None
+        if relpath is None:
+            relpath = Path('.')
+
+        authinfo = self.get_authinfo()
+        full_path = Path(self.get_remote_path()) / relpath
+        computer_label = self.computer.label if self.computer is not None else ''
+
+        with authinfo.get_transport() as transport:
+            if not transport.path_exists(str(full_path)):
+                exc_message = (
+                    f'The required remote folder {full_path} on Computer <{computer_label}> ' 'does not exist.'
+                )
+                raise FileNotFoundError(exc_message)
+
+            if method not in ('du', 'lstat'):
+                raise NotImplementedError(
+                    f'Specified method `{method}` for evaluating the size on disk not implemented.'
+                )
+
+            if method == 'du':
+                try:
+                    total_size: int = self._get_size_on_disk_du(full_path, transport)
+
+                except (RuntimeError, NotImplementedError):
+                    # NotImplementedError captures the fact that, e.g., FirecREST does not allow for `exec_command_wait`
+                    lstat_warn = (
+                        'Problem executing `du` command. Will return total file size based on `lstat`. '
+                        'Take the result with a grain of salt, as `lstat` does not consider the file system block '
+                        'size, but instead returns the true size of the file contents in bytes, which is usually '
+                        'smaller than the actual space requirements on disk.'
+                    )
+
+                    _logger.warning(lstat_warn)
+                    method = 'lstat'
```

Comment:
Yeah, I initially didn't re-assign and return it, but it made the user report nicer, as I could echo the method that was eventually used, because it could be that `du` fails, and then `lstat` is being used. Now that I'm thinking about it, I'm wondering, if I should even switch it internally, or just let the command fail if `du` is selected and fails, and tell the user to try it with the other method? What would you, as a user, prefer?

Also, having the return inside each try-except leads to duplication of the code snippet where I check if `return_bytes` is set. Not sure if there's a way around it, tbh, but for me, either solution works.